### DISCUSSION TITLE
Best way to do show back for subscription owners

### DISCUSSION THREAD

**Discussion by kylgrn** [microsoft/finops-toolkit#697] [Q&A]
Hello, wondering what the best approach would be using this toolkit to develop a standardized report that will be accessible by subscription owners to monitor their own costs. They shouldn't be able to see the cost data for neighboring subscriptions.

**Comment by MS-mikeo** [comment]
I did some testing on this and it looks like you could accomplish this using Row Level Security.  https://learn.microsoft.com/en-us/fabric/security/service-admin-row-level-security

I created a new Role for subscription 1 in my environment that contains 5 subscriptions.   The subscription filter that is typically used in the FinOps hubs report is SubAccountNameUnique.  I created a role on the CostDetails table where SubAccountNameUnique equals _Subscription name (subscriptionID_. (Example - **ME-Subscription1 (xxxd-axxc-4xxx2b-8axx-weffcsdexxxx**).

![image](https://github.com/microsoft/finops-toolkit/assets/130005316/93abdcdc-b65d-4ab0-8ce4-e4d8c71c4ba2)

When I look at the Subscription view without the view enabled, I see all subscriptions. 
![image](https://github.com/microsoft/finops-toolkit/assets/130005316/14bfc7a7-c986-40a5-8f2a-046af8f36e3c)

Now when I enable View as and then the role Subscription1, I only see data where the rows in CostDetails contain the SubAccountNameUnique specified for that role.

![image](https://github.com/microsoft/finops-toolkit/assets/130005316/c1e73b95-4d34-4729-82a0-a6ce3a3ee5a1)
![image](https://github.com/microsoft/finops-toolkit/assets/130005316/80a55098-63a3-4d12-af47-aa46eb6ced03)

The last step is then to publish the report and then assign this role.

![image](https://github.com/microsoft/finops-toolkit/assets/130005316/7768b5f5-56a9-44f4-9893-8e89daad5a19)
